polars/08_working_with_columns.py

```diff
@@ -0,0 +1,605 @@
+def _(mo):
+    mo.md(r"""At first, you may think about using a `for` loop:""")
+    return
+
+
+@app.cell
+def _(ext_temp_data, pl):
+    _result = ext_temp_data
+    for col_name in ["Air", "Process"]:
+        _result = _result.with_columns(
+            (abs(pl.col(f"Rolling mean {col_name} temperature") - pl.col(f"{col_name} temperature")))
+                .round(2).alias(f"Delta {col_name} temperature")
+        )
+    _result
+    return (col_name,)
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Using a `for` loop is functional, but not scalable, as each expression needs to be defined in an iteration and executed serially. Instead we can use a generator in Python to programmatically create all expressions at once. In conjunction with the `with_columns` context, we can take advantage of parallel execution of computations and query optimization from Polars.""")
+    return
+
+
+@app.cell
+def _(ext_temp_data, pl):
+    def delta_expressions(colnames: list[str]) -> pl.Expr:
+        for col_name in colnames:
+            yield (abs(pl.col(f"Rolling mean {col_name} temperature") - pl.col(f"{col_name} temperature"))
+                            .round(2).alias(f"Delta {col_name} temperature"))
+
+
+    ext_temp_data.with_columns(delta_expressions(["Air", "Process"]))
+    return (delta_expressions,)
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""
+        ## More flexible column selections
+
+        For more flexible column selections, you can use column selectors from `selectors`. Column selectors allow for more expressiveness in the way you specify selections. For example, column selectors can perform the familiar set operations of union, intersection, difference, etc. We can use the union operation with the functions `string` and `ends_with` to select all string columns and the columns whose names end with "`_high`":
+        """
+    )
+    return
+
+
+@app.cell
+def _(data):
+    import polars.selectors as cs
+
+    data.select(cs.string() | cs.ends_with("F"))
+    return (cs,)
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Likewise, you can pick columns based on the category of the type of data, offering more flexibility than the `col` function. As an example, `cs.numeric` selects numeric data types (including `pl.Float32`, `pl.Float64`, `pl.Int32`, etc.) or `cs.temporal` for all dates, times and similar data types.""")
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""
+        ### Combining selectors with set operations
+
+        Multiple selectors can be combined using set operations and the usual Python operators:
+
+
+        | Operator |       Operation      |
+        |:--------:|:--------------------:|
+        | `A \| B`   | Union                |
+        | `A & B`    | Intersection         |
+        | `A - B`    | Difference           |
+        | `A ^ B`    | Symmetric difference |
+        | `~A`       | Complement           |
```

Comment:
```suggestion
        | Operator |       Operation      |
        |:--------:|:--------------------:|
        | \( A \mid B \)   | Union                |
        | \( A \cap B \)   | Intersection         |
        | \( A - B \)      | Difference           |
        | \( A \oplus B \) | Symmetric difference |
        | \( \sim A \)     | Complement           |
```

Better visual appeal (with LaTeX in markdown)?